File: LoanApplicationWebAPI/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LoanApplicationWebAPI.Models
{
    public class Customer : User
    {
        [StringLength(12, MinimumLength = 12)]
        public string AadharId { get; set; }

        [StringLength(10, MinimumLength = 10)]
        public string PanId { get; set; }

        [Required]
        [Phone]
        public string ContactNumber { get; set; }

        public DateTime DateOfBirth { get; set; }
        public string City { get; set; }

        [JsonIgnore]
        public ICollection<LoanApplication>? LoanApplications { get; set; }
    }
}

File: src/app/Models/customer.ts
// src/app/Models/customer.ts
import { User, UserRole } from './user';

// Export UserRole immediately (so other files can import it from customer if needed)
export type { UserRole };

export interface Customer extends User {
  aadharId: string;        // 12 chars
  panId: string;           // 10 chars
  contactNumber: string;   // Phone number
  dateOfBirth: string;     // ISO string for backend
  city: string;
}
